ocpp1.6/central_system.go

```diff
@@ -374,10 +383,10 @@ func (cs *centralSystem) SendRequestAsync(clientId string, request ocpp.Request,
 		return fmt.Errorf("unsupported action %v on central system, cannot send request", featureName)
 	}
 
-	cs.insertCallback(clientId, callback)
+	cs.callbackQueue.Queue(clientId, callback)
 
 	if err := cs.server.SendRequest(clientId, request); err != nil {
-		_, _ = cs.popCallback(clientId)
+		_, _ = cs.callbackQueue.Dequeue(clientId)
```

Comment:
There's an issue here: if there was already a callback for that client in the queue, you will pop the first element in the queue and not the one, that was just added, e.g.:

| cb1 | cb2 | cb3| cb4 |

in line 386 you would add `cb4`, but when calling the Dequeue you would be removing `cb1` instead, which messes up all other callbacks. 

The only possible solution I see, is to enqueue the callback only after checking the error.
This leads to another problem -> see my other comment.

Reply:
Yeah of course, you're right, that would need to be a pop.